EntityFramework.Exceptions.Tests/OracleTests.cs

```diff
@@ -7,17 +9,46 @@ namespace EntityFramework.Exceptions.Tests
 {
     class OracleTests : DatabaseTests, IClassFixture<OracleTestContextFixture>
     {
-        public OracleTests(OracleTestContextFixture fixture) : base(fixture.Context)
+        OracleTests(OracleTestContextFixture fixture) : base(fixture.Context)
         {
         }
+
+        private async Task DeleteOfProductInSaleShouldThrowReferenceConstraintException()
+        {
+            var author = new Author
+            {
+                Name = "William Shakespeare"
+            };
+            var book = new Book
+            {
+                Name = "Hamlet",
+                Author = author
+            };
+            Context.Books.Add(book);
+            await Context.SaveChangesAsync();
+            CleanupContext();
+
+            var newAuthor = Context.Authors.Find(author.Id);
+            Context.Authors.Remove(newAuthor);
+            Assert.Throws<ReferenceConstraintException>(() => Context.SaveChanges());
+            await Assert.ThrowsAsync<ReferenceConstraintException>(() => Context.SaveChangesAsync());
+        }
+
+        private void CleanupContext()
```

Comment:
Can you move it to the `DatabaseTests` to the class? It could be used in the `Dispose` method too.